src/Nethermind/Nethermind.AccountAbstraction/Executor/UserOperationSimulator.cs

```diff
@@ -117,35 +141,52 @@ public class UserOperationSimulator : IUserOperationSimulator
             else
             {
                 userOperation.AccessList = validationAccessList;
+                userOperation.AddressesToCodeHashes = addressesToCodeHashes;
                 userOperation.AlreadySimulated = true;
             }
 
-            return ResultWrapper<Keccak>.Success(userOperation.Hash);
+            return ResultWrapper<Keccak>.Success(userOperation.RequestId!);
         }
 
-        private (bool success, UserOperationAccessList accessList, string? error) SimulateValidation(
+        private (bool success, UserOperationAccessList accessList, IDictionary<Address, Keccak> addressesToCodeHashes, string? error) SimulateValidation(
```

Comment:
4 return parameters - maybe better to put them into a class?

Reply:
I would refactor this

Reply:
done